src/comparison_tools/PairwiseCompare.py

```diff
@@ -189,33 +201,59 @@ def inter_comparisons(self):
                 self.__warn_empty_comparisons(_comparison_type_name="Inter Comparisons")
                 continue
 
+            if len(self.__filtered_antehoc_col_idx) < len(self.__antehoc_group_cols):
+                filtered_apair = tuple(
+                    self.__get_group_column_element(
+                        apair[group_idx], self.__filtered_antehoc_col_idx
+                    )
+                    for group_idx in range(2)
+                )
+
             # Iterate through each well group cartesian product and save the data
             for ppair in comparison_key_product:
 
                 if self.__contains_match(ppair):
                     continue
 
-                ppair0 = ppair[0]
-                ppair1 = ppair[1]
+                ppair = tuple(
+                    [
+                        (item,) if not isinstance(item, tuple) else item
+                        for item in ppair[:2]
+                    ]
+                )
 
-                # Avoids a future deprecation in the pandas get_group method
-                if not isinstance(ppair0, tuple) or not isinstance(ppair1, tuple):
-                    ppair0 = (ppair0,)
-                    ppair1 = (ppair1,)
+                ppair0, ppair1 = ppair
 
                 self.__comparator(
                     group0df.get_group(ppair0), group1df.get_group(ppair1)
                 )
 
-                self.__comparator.save_groups(
-                    self.__antehoc_group_cols,
-                    **dict(zip(self.__antehoc_group_names, apair)),
-                )
-
-                self.__comparator.save_groups(
-                    self.__posthoc_group_cols,
-                    **dict(zip(self.__posthoc_group_names, ppair)),
-                )
+                if len(self.__filtered_antehoc_col_idx) < len(
+                    self.__antehoc_group_cols
+                ):
+                    self.__comparator.save_groups(
+                        self.__get_group_column_element(
+                            self.__antehoc_group_cols, self.__filtered_antehoc_col_idx
+                        ),
+                        **dict(zip(self.__antehoc_group_names, filtered_apair)),
+                    )
+
+                if len(self.__filtered_posthoc_col_idx) < len(
+                    self.__posthoc_group_cols
+                ):
+                    filtered_ppair = tuple(
+                        self.__get_group_column_element(
+                            ppair[group_idx], self.__filtered_posthoc_col_idx
+                        )
+                        for group_idx in range(2)
+                    )
+
+                    self.__comparator.save_groups(
+                        self.__get_group_column_element(
+                            self.__posthoc_group_cols, self.__filtered_posthoc_col_idx
+                        ),
+                        **dict(zip(self.__posthoc_group_names, filtered_ppair)),
+                    )
```

Comment:
Is this code fixing the same and different columns issue?

Reply:
Exactly, and it's making sure the output is in the correct format.